vunit/ui/__init__.py

```diff
@@ -795,6 +818,11 @@ def _main_run(self, post_run):
         Main with running tests
         """
         simulator_if = self._create_simulator_if()
+        try:
+            simulator_if.set_global_compile_options(self._global_compile_options)
```

Comment:
Why is every python exception caught and not the ones you care about? This could mask potentially important issues in the code that would make debugging difficult

Reply:
Also, it could be written more succinctly using `with suppress(Exception):` (and `from contextlib import suppress`)

Reply:
Do you know why this is catching all errors though? My concern is things like `KeyError` somewhere below that should be raised and might cause confusing errors later. My opinion is that this should only catch what it cares about to avoid masking VUnit bugs that affect the user.

Reply:
I believe this was added because we didn't care if it failed which is probably not the behavior we'd actually want in production quality code. I'll have to dig in and see what possible failures actually exist and see if we even care to catch any of them.